File: saveDataDemo/app/src/main/java/edu/cs4730/SaveData/DataViewModel.java
package edu.cs4730.SaveData;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

public class DataViewModel extends AndroidViewModel {
    private MutableLiveData<Integer> d4;

    public DataViewModel(@NonNull Application application) {
        super(application);
        d4 = new MutableLiveData<Integer>();
        d4.setValue(0);
    }

    public LiveData<Integer> getData() {
        return d4;
    }

    public int getValue() {
        return d4.getValue().intValue();

    }

    public void increment() {
        d4.setValue(d4.getValue() + 1);
    }
}

File: saveDataDemo_kt/app/src/main/java/edu/cs4730/savedatademo_kt/DataViewModel.kt
package edu.cs4730.savedatademo_kt

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.LiveData

class DataViewModel(application: Application) : AndroidViewModel(application) {
    var d4: MutableLiveData<Int> = MutableLiveData()
    val data: LiveData<Int>
        get() = d4
    val value: Int
        get() = d4.value!!

    fun increment() {
        d4.value = d4.value!! + 1
    }

    init {
        d4.value = 0
    }
}